test_src/sljitTest.c

```diff
@@ -11272,51 +11291,59 @@ static void test90(void)
 	/* Test reverse four bytes. */
 	executable_code code;
 	struct sljit_compiler* compiler = sljit_create_compiler(NULL, NULL);
-	sljit_sw buf[12];
-	sljit_s32 ibuf[5];
+	sljit_sw buf[10];
+	sljit_s32 ibuf[7];
 	sljit_s32 i;
 
 	if (verbose)
 		printf("Run test90\n");
 
 	FAILED(!compiler, "cannot create compiler\n");
 
-	for (i = 0; i < 12; i++)
+	for (i = 0; i < 10; i++)
 		buf[i] = -1;
-	for (i = 0; i < 5; i++)
+	for (i = 0; i < 7; i++)
 		ibuf[i] = -1;
 
 	sljit_emit_enter(compiler, 0, SLJIT_ARGS2(VOID, W, W), 5, 5, 0, 0, 2 * sizeof(sljit_sw));
 
+	/* zero extend any integer */
 	sljit_emit_op1(compiler, SLJIT_MOV, SLJIT_R0, 0, SLJIT_IMM, WCONST(0xffffa1b2c3d4, 0xa1b2c3d4));
 	sljit_emit_op1(compiler, SLJIT_REV_U32, SLJIT_R0, 0, SLJIT_R0, 0);
 	/* buf[0] */
 	sljit_emit_op1(compiler, SLJIT_MOV, SLJIT_MEM1(SLJIT_S0), 0, SLJIT_R0, 0);
 
-	sljit_emit_op1(compiler, SLJIT_MOV, SLJIT_R2, 0, SLJIT_IMM, WCONST(0xffffa1b2c3d4, 0xa1b2c3d4));
```

Comment:
All similar tests should be kept.

Reply:
This test is still doing the same, with the only caveat that is now no longer undefined, because the value stored in the register matches what the operation (SLJIT_REV_S32) expects regarding sign extension of the upper 32bit.

The special case were that is not undefined is now covered by the improved case12 that does so by using a MEM destination and ensuring the relevant value was written and that write didn't overflow.